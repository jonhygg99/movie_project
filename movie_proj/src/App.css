
.App {
  background: #282c34;
	position: absolute;
	top:0;
	bottom: 0;
	left: 0;
	right: 0;
}

.App-logo, .App-logo-loading {
  animation: App-logo-spin infinite 20s linear;
  pointer-events: none;
}

.App-logo {
  height: 50px;
}

.App-logo-loading {
  height: 50%;
  width: 20%;

}

.Loading {
  display: flex;
  align-items: center;
  justify-content: center;
  background: #282c34;
	position: absolute;
	top:0;
	bottom: 0;
	left: 0;
	right: 0;
	margin: auto;
}

.App-header, .App-footer {
  background-color: #282c34;
  display: flex;
  align-items: center;
  justify-content: center;
  color: white;
  padding: 10px;
}

.App-header > h1 {
  font-size: 20px;
}

.App-body{
  background-color: #414d64;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  padding: 10px;
}

.Search-bar {
  animation: search-big-square 4s ease;/*need to expand*/
  width: 30%;
  text-align: center;
  font-family: sans-serif;
  font-weight: bold;
  font-size: medium;
  padding: 15px 22px;
  margin: 10px 5px;
  border: 6px solid #8842d5;
  border-radius: 40px;  
  background-color: rgb(224, 224, 224);
  color: black;
  resize: none;  
}

.Title-section {
  text-decoration: underline;
  color:rgb(211, 211, 211);
}

.Movie-view {
  text-align: center;
}

.Image-movie {
    border-radius: 8px;
    width: 25%;
    min-width: 200px;
    margin-left: 2%;
    margin-bottom: 2%;
  }

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@keyframes search-big-square {
  from {
    width: 30%;
  }
  to {
    widht: 50%;
  }
}