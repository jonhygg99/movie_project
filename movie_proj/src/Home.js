import React from 'react';
import logo from './logo.svg';
import { PopularSelections } from './PopularSelections'
import { API_SEARCH, API_KEY, API_POPULAR, language } from './constants'
import { getSearchUrl } from './utils'
import { MovieView } from './MovieView'
import { debounce } from 'lodash'

export class Home extends React.Component {
  constructor(props) {
    super(props);
			this.state = { 
        selection: 'popular?',
        userInput: '',
        items: [],
        isLoading: true,
        page: 1,
        max_pages: 1
      };
       this.handleUserInput = this.handleUserInput.bind(this)
       this.decrementPage = this.decrementPage.bind(this)
       this.incrementPage = this.incrementPage.bind(this)
       this.timeout = null
  }

  componentDidMount() {
    fetch(API_POPULAR)
    .then(response => response.json())
    .then(json => {
      this.setState({ 
        items: json.results, 
        isLoading: false,
        max_pages: json.total_pages
       });
    });
  }

  componentDidUpdate(prevProps, prevState) {
    const delay = 500
    var timeoutID
    if (prevState.userInput !== this.state.userInput || 
      prevState.page !== this.state.page) {
        if (prevState.userInput !== this.state.userInput)
          this.setState({page: 1});
        if (this.state.userInput === '' && 
        (prevState.userInput !== this.state.userInput || this.state.page === 1) ) {
          this.timeout = setTimeout( () =>
          fetch(API_POPULAR)
          .then(response => response.json())
          .then(json => {
            this.setState({
              selection: 'popular?',
              items: json.results, 
              isLoading: false,
              max_pages: json.total_pages
             });
          }), delay);
        } else if (this.state.userInput === '' && (prevState.page !== this.state.page)) {
          this.timeout = setTimeout( () =>
          fetch(API_SEARCH + this.state.selection + API_KEY + language + '&page=' + this.state.page)
          .then(response => response.json())
          .then(json => {
            this.setState({ 
              items: json.results, 
              isLoading: false,
              max_pages: json.total_pages
             });
          }), delay);
        } else {
          this.timeout = setTimeout( () =>
          fetch(getSearchUrl(this.state.userInput, this.state.page))
          .then(response => response.json())
          .then(json => {
          this.setState({ 
            selection: '',
            items: json.results, 
            isLoading: false,
            max_pages: json.total_pages });
          }), delay);
        }
      }
      //clearTimeout(this.timeout)

  }

  render() {
    var { items, isLoading } = this.state

    // if (isLoading) {
    //   return <div className="Loading">
    //             <img src={logo} className="App-logo-loading" alt="logo" />
    //             </div>
    // } else {
       return (
         <div className="App-body">
          <input className="Search-bar" type="text" 
            placeholder="Search movies..." 
            onFocus={(e) => e.target.placeholder = ""} 
            onBlur={(e) => e.target.placeholder = "Search movies..."}
            onChange={this.handleUserInput}
            value={this.state.userInput}/>
        <PopularSelections value={this.state.selection}/>
        <h1>Input: {this.state.userInput}</h1>
        <h1>Pages {this.state.max_pages}</h1>
        <div className="changePage">
          <button className="Selections" id="decrement"onClick={this.decrementPage}>Previous</button>
          <h1 className="changePageNumber">{this.state.page}</h1>
          <button className="Selections" id="increment" onClick={this.incrementPage}>Next</button>
        </div>
        <MovieView items={items}/>
        </div>
      );
    //}
  }

  handleUserInput(e) {
    // Clear the timeout if it has already been set.
    // This will prevent the previous task from executing
    // if it has been less than <MILLISECONDS>
    clearTimeout(this.timeout);

    const value = e.target.value
    this.setState({
      userInput: e.target.value
    });

    // Make a new timeout set to go off in 800ms
    //this.timeout = setTimeout(function () {
    //  setInput()
    //}, 500);
  }
  decrementPage () {
    const newPage = this.state.page - 1
    this.setState({ page: newPage < 1 ? 1 : newPage })
  }
  incrementPage () {
    const newPage = this.state.page + 1
    this.setState({ page: newPage > this.state.max_pages ? this.state.page : newPage})
  }
}

export default Home;
