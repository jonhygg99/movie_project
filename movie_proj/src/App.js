import React from 'react';
import logo from './logo.svg';
import './App.css';

const API_KEY = 'a2fd13908e7ca798960594a0d61175e6'
const API_POPULAR = 'https://api.themoviedb.org/3/movie/popular?api_key=' + API_KEY

class App extends React.Component {
  constructor(props) {
    super(props);
			this.state = { 
        userInput: '',
        data_movie: []
      };
   		this.handleUserInput = this.handleUserInput.bind(this);
  }

  componentDidMount() {
    fetch(API_POPULAR)
    .then(response => response.json())
    .then(data => this.setState({ data_movie: data.data_movie }));
  }
  

  handleUserInput(e) {
    this.setState({
      userInput: e.target.value
    });
  }
  
  render() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h1>Movie Database</h1>
      </header>
      <body className="App-body">
       <input className="Search_bar" type="text" placeholder="Search movies..."  
          onChange={this.handleUserInput}
          value={this.state.userInput}/>
        <h1>{this.state.userInput}</h1>
       <p>Search for something new</p>
       <p>{API_POPULAR}</p>
      </body>
      <footer className="App-footer">
        <div>
        <p>Lorem ipsum dolor sit amet consectetur, adipiscing elit ornare nisl.</p>
        </div>

      </footer>
    </div>
  );
  }
}

export default App;
