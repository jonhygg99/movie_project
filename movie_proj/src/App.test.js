import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Home from './Home'
import { shallow } from 'enzyme'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('page value', () => {
  const wrapper = shallow(<Home />)
  const countState = wrapper.state().page
  expect(countState).toEqual(1)
});

it('next page', () => {
  const wrapper = shallow(<Home />)
  //const text = wrapper.find('p').text()
  const incrementPage = wrapper.find('button.incrementPage')
  incrementPage.simulate('click')
  const countState = wrapper.state().page
  expect(countState).toEqual(2)
  //expect(text).toEqual('1')
});