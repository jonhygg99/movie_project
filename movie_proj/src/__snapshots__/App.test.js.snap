// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="Movie-view"
>
  <h1>
    No movie found, try another name
  </h1>
</div>
`;
